sql/core/src/test/scala/org/apache/spark/sql/errors/QueryParsingErrorsSuite.scala

```diff
@@ -27,9 +27,10 @@ class QueryParsingErrorsSuite extends QueryTest with QueryErrorsSuiteBase {
     validateParsingError(
       sqlText = "SELECT * FROM t1 NATURAL JOIN LATERAL (SELECT c1 + c2 AS c2)",
       errorClass = "UNSUPPORTED_FEATURE",
+      errorSubClass = Some("LATERAL_NATURAL_JOIN"),
       sqlState = "0A000",
       message =
-        """The feature is not supported: "LATERAL" join with "NATURAL" join.(line 1, pos 14)
+        """The feature is not supported: NATURAL join with LATERAL correlation.(line 1, pos 14)
```

Comment:
In fact, you reverted my recent changes of using double quotes in error messages: https://github.com/apache/spark/pull/36259. Could you add double quotes for SQL statements in the json file, please.

Reply:
I saw that. But the intent was to quote/double quote **parameters**. We use unquoted keywords everywhere in the docs e.g. **WHERE clause**

Reply:
Users don't care how we form error messages either via passing SQL statements or by inlining them into templates in the json file. Independently from the internal mechanism, the output should be consistent I do believe either to output all SQL statement in double quotes or w/o them everywhere in error messages.

Reply:
Let's only introduce sub-classes by this PR and keep the same quoting. We can discuss quoting separately later. I think new quoting rules is not related to sub-classes, and I don't see any reasons to mix unrelated changes here.

Reply:
@HyukjinKwon @cloud-fan Since you reviewed my PR https://github.com/apache/spark/pull/36259, are you ok with reverting the chages?

Reply:
Here is the PR https://github.com/apache/spark/pull/36359